Common/Securities/Positions/PositionManager.cs

```diff
@@ -0,0 +1,202 @@
+                            }
+                        }
+                    }
+                    else if (args.Action == NotifyCollectionChangedAction.Remove)
+                    {
+                        if (Groups.Contains(key))
+                        {
+                            security.Holdings.QuantityChanged -= HoldingsOnQuantityChanged;
+                            if (security.Invested)
+                            {
+                                // only trigger group resolution if we had holdings in the removed security
+                                _requiresGroupResolution = true;
+                            }
+                        }
+                    }
+                }
+            };
+        }
+
+        /// <summary>
+        /// Gets the <see cref="IPositionGroup"/> matching the specified <paramref name="key"/>. If one is not found,
+        /// then a new empty position group is returned.
+        /// </summary>
+        public IPositionGroup this[PositionGroupKey key] => Groups[key];
+
+        /// <summary>
+        /// Creates a position group for the specified order, pulling
+        /// </summary>
+        /// <param name="order">The order</param>
+        /// <returns>A new position group matching the provided order</returns>
+        public IPositionGroup CreatePositionGroup(Order order)
+        {
+            IPositionGroup group;
+            var positions = order.CreatePositions(_securities).ToList();
+            if (!_resolver.TryGroup(positions, out group))
+            {
+                throw new InvalidOperationException($"Unable to create group for order: {order.Id}");
+            }
+
+            return group;
+        }
+
+        /// <summary>
+        /// Applies the order fill without running the position group resolver. This aims to ensure we're
+        /// keeping track of used buying power within a single time step to handle multiple calls to SetHoldings
+        /// in the same time step. At the end of the time step
+        /// </summary>
+        /// <param name="fill">The fill event -- this should only be events that change security holdings</param>
+        public void ProcessFill(OrderEvent fill)
+        {
+            if (fill.Status.IsFill())
+            {
+                ResolvePositionGroups();
+            }
+        }
+
+        /// <summary>
+        /// Resolves the algorithm's position groups from all of its holdings
+        /// </summary>
+        public void ResolvePositionGroups()
+        {
+            if (_requiresGroupResolution)
+            {
+                _requiresGroupResolution = false;
+                // TODO : Replace w/ special IPosition impl to always equal security.Quantity and we'll
+                // use them explicitly for resolution collection so we don't do this each time
+
+                Groups = ResolvePositionGroups(new PositionCollection(_securities
+                    .Where(kvp => kvp.Value.Invested).ToImmutableDictionary(
```

Comment:
`.Where(kvp => kvp.Value.Invested)` Will remove uninvested securities which seems different from the behavior at `securities.CollectionChanged +=` where we add all securities with the default position group?